### DISCUSSION TITLE
SSL/TLS and Session Cache Status Stats

### DISCUSSION THREAD

**Discussion by TheOldMan2000** [netdata/netdata#13395] [Ideas]
Hello Team Netdata,

Is it possible for Netdata to monitor/chart the stats that appear at the bottom of the WHM/Cpanel Apache Status page (WHM Home > Server Status > Apache Status ):

````
SSL/TLS Session Cache Status:
--
cache type: SHMCB, shared memory: 1024000 bytes, current entries: 64subcaches: 32, indexes per subcache: 177time left on oldest entries' objects: avg: 107 seconds, (range: 1...286)index usage: 1%, cache usage: 1%total entries stored since starting: 252total entries replaced since starting: 0total entries expired since starting: 188total (pre-expiry) entries scrolled out of the cache: 0total retrieves since starting: 0 hit, 0 misstotal removes since starting: 0 hit, 0 miss

SSL/TLS Session Cache Status:
cache type: SHMCB, shared memory: [1024000](tel:1024000) bytes, current entries: 64
subcaches: 32, indexes per subcache: 177
time left on oldest entries' objects: avg: 107 seconds, (range: 1...286)
index usage: 1%, cache usage: 1%
total entries stored since starting: 252
total entries replaced since starting: 0
total entries expired since starting: 188
total (pre-expiry) entries scrolled out of the cache: 0
total retrieves since starting: 0 hit, 0 miss
total removes since starting: 0 hit, 0 miss
````

**Comment by ilyam8** [comment]
Hi, @TheOldMan2000. So it is additional info you get from [mod_status](https://httpd.apache.org/docs/2.4/mod/mod_status.html)? The data sample you posted - is it all data you get from "WHM Home > Server Status > Apache Status" or do you show only part related to Session Cache Status?

Anyway, it is easy to implement (update apache collector) if somebody will:

 - tell how to set up a local Apache instance with [SSLSessionCache](https://cwiki.apache.org/confluence/display/HTTPD/SSLSessionCache).
 - tell what metrics are worth collecting.

**Reply by TheOldMan2000** [reply]
Hi @ilyam8 

Yes I was just showing the relevant bit at the very bottom of the page. It shows up at the bottom if you use a browser to go to the server-status page too, not just a WHM/Cpanel feature. 

E.g. 

example.com/server-status
example.com/server-status?auto
example.com/server-status.refresh=2

The exact output depends on what you have enabled:

https://httpd.apache.org/docs/current/mod/mod_ssl.html#sslsessioncache
https://httpd.apache.org/docs/current/mod/mod_ssl.html#sslstaplingcache

![2AC1D0BF-07E4-4A79-8484-DDF951B75245](https://user-images.githubusercontent.com/13329789/180043631-1d63018f-7caa-4b96-8e5d-206da9c40686.jpeg)
![514E7AB4-4BC0-47FE-84BB-BC6AB1F93739](https://user-images.githubusercontent.com/13329789/180043649-c3493e47-25eb-4085-868e-7e20f6ccc83b.jpeg)

**Reply by ilyam8** [reply]
@TheOldMan2000 consider creating a [feature request](https://github.com/netdata/netdata/issues/new/choose).
